aten/src/ATen/THLongStorageView.h

```diff
@@ -53,22 +53,33 @@ class THLongStorageView {
         break;
     }
 
-    if(zero_dim_to_one && ref.size() == 0) {
+    if (zero_dim_to_one && ref.size() == 0) {
       // make storage of size 0 actually a 1-length storage with 1 element
       // so that our 0-dim tensors get allocated as 1-dim inside TH
+
       one = 1;
-      storage.data_ptr = {&one, kCPU}; // non-owning
-      storage.size = 1;
+      storage.set_pImpl(new StorageImpl(
```

Comment:
This comment is not requesting a code change, it's just a "for your information."

Because this is a constructor, you could have avoided the need for `set_pImpl` by simply calling the correct constructor on `storage` at the very beginning. Obviously the `StorageImpl` to allocate in this case is somewhat complex, so you'd add a helper function to actually compute and return it.